sources/lib/structures/typed.mli

```diff
@@ -26,129 +26,260 @@
+(** Type alias for annoted typed atoms. *)
 
 and 'a tatom =
   | TAtrue
+  (** The [true] atom *)
   | TAfalse
-  | TAeq of ('a tterm, 'a) annoted list
-  | TAdistinct of ('a tterm, 'a) annoted list
-  | TAneq of ('a tterm, 'a) annoted list
-  | TAle of ('a tterm, 'a) annoted list
-  | TAlt of ('a tterm, 'a) annoted list
-  | TApred of ('a tterm, 'a) annoted * bool (* true <-> negated *)
+  (** The [false] atom *)
+  | TAeq of 'a atterm list
+  (** Equality of a set of typed terms. *)
+  | TAdistinct of 'a atterm list
+  (** Disequality. All terms in the set are pairwise distinct.
+      TODO: check interpretation ? *)
+  | TAneq of 'a atterm list
+  (** Equality negation: at least two elements in the list
+      are not equal.
+      TODO: check interpretation. *)
+  | TAle of 'a atterm list
+  (** Arithmetic ordering: lesser or equal. Chained on lists of terms.
+      TODO: check interpretation ? *)
+  | TAlt of 'a atterm list
+  (** Strict arithmetic ordering: less than. Chained on lists of terms.
+      TODO: check interpretation ? *)
+  | TApred of 'a atterm * bool
+  (** Term predicate, negated if the boolean is true.
+      [TApred (t, negated)] is satisfied iff [t <=> not negated] *)
+(** Typed atoms. *)
 
 and 'a quant_form = {
-  (* quantified variables that appear in the formula *)
-  qf_bvars : (Symbols.t * Ty.t) list ;
-  qf_upvars : (Symbols.t * Ty.t) list ;
-  qf_triggers : (('a tterm, 'a) annoted list * bool) list;
-  qf_hyp : ('a tform, 'a) annoted list;
-  qf_form : ('a tform, 'a) annoted
+  qf_bvars : (Symbols.t * Ty.t) list;
+  (** Quantified variables (with their types) that appear in the formula.
+      TODO: reword this to say whether those variables appear int he formula,
+            or whether they are the variables quantified by the formula (in the
+            inner formula [qf_form]) *)
+  qf_upvars : (Symbols.t * Ty.t) list;
+  (** TODO: what are thoses ? *)
+  qf_triggers : ('a atterm list * bool) list;
+  (** Triggers associated wiht the formula.
+      TODO: what is the boolean ? *)
+  qf_hyp : 'a atform list;
+  (** TODO: what is this field ? *)
+  qf_form : 'a atform;
+  (** The quantified formula. *)
 }
+(** Quantified formulas. *)
+
+and 'a atform = ('a tform, 'a) annoted
+(** Type alias for typed annoted formulas. *)
 
 and 'a tform =
-  | TFatom of ('a tatom, 'a) annoted
-  | TFop of oplogic * (('a tform, 'a) annoted) list
+  | TFatom of 'a atatom
+  (** Atomic formula. *)
+  | TFop of oplogic * 'a atform list
+  (** Application of logical operators. *)
   | TFforall of 'a quant_form
+  (** Universal quantification. *)
   | TFexists of 'a quant_form
+  (** Existencial quantification. *)
   | TFlet of (Symbols.t * Ty.t) list *
-             (Symbols.t * 'a tlet_kind) list * ('a tform, 'a) annoted
-  | TFnamed of Hstring.t * ('a tform, 'a) annoted
+             (Symbols.t * 'a tlet_kind) list *
+             'a atform
+  (** Let binding.
+      TODO: what is in the first list ? *)
```

Comment:
Yeah, but a list of binders that bind what ? each symbol is associated to a type, so it does seem not bind type variables, and does not bind terms (because there are no terms in the list, plus terms are quantified in the second list).